### PR TITLE
resources: refactor UI serializer

### LINKED ISSUE #242: landing page: publication date should be localized
## Describe the bug

The publication date is not localized on the publication date.

## Steps to Reproduce

1. Go to the landing page of any record (https://127.0.0.1:5000/records/<recid>)

## Expected behavior

The text should read like in the expected screenshot below. Note, do not add the version or make a label around the date. It's only the actual text which is wrong. The styling is fine.

## Screenshots (if applicable)

Current:

Screenshot 2020-11-02 at 14 53 28

Expected:

Screenshot 2020-11-02 at 14 55 20

### LINKED ISSUE #257: resources: refactor serialisers
# Issues

- We have a single ``serializers.py`` and we need many more serializers, so we need to create a package.
- Marshmallow-Utils have been udpate with new fields.
- The current UI serializer won't scale if we add more stuff.
- Vocabulary code in UI serializer needs to use the vocabulary API instead.

### PR DESCRIPTION
**Pull request by lnielsen**
Resolves #257 
Closes #242

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -70,12 +71,11 @@
     'invenio-communities>=2.1.1,<3.0.0',
     'invenio-drafts-resources>=0.5.0,<0.6.0',
     'invenio-formatter[badges]>=1.1.0a1,<2.0.0',
-    "invenio-i18n>=1.2.0",
+    'invenio-i18n>=1.2.0',
     'invenio-records>=1.4.0a4,<2.0.0',
     'invenio-records-files>=1.2.1,<2.0.0',
     'invenio-records-ui>=1.2.0a1,<2.0.0',
     'invenio-previewer>=1.2.1,<2.0.0',
-    'marshmallow-utils>=0.2.2,<0.3.0',
```
Comment (lnielsen) [inline setup.py:78]: Specified by Invenio-Records-Resources

File: setup.py [hunk 2]
```diff
@@ -62,6 +62,7 @@
 
 install_requires = [
     'arrow>=0.17.0',
+    'babel-edtf>=1.0.0',
```
Comment (lnielsen) [inline setup.py:65]: Minor: Remove (specified by Marshmallow-Utils)

File: invenio_rdm_records/vocabularies/resource_type.py [hunk 3]
```diff
@@ -29,17 +31,19 @@ def key(self, row):
         """
         return (row.get('type'), row.get('subtype', ''))
 
+    def get_entry_by_dict(self, dict_key):
```
Comment (lnielsen) [inline invenio_rdm_records/vocabularies/resource_type.py:34]: Allows to query by a key.

**Comment by lnielsen** [comment]
Example of object ui key:

```
  "ui": {
          "creators": {
            # A list of affiliations to easily display the list.
            "affiliations": [
              [
                1,
                "Lopez Ltd"
              ],
              ...
            ], 
            # Creators annotated with affiliation index.
            "creators": [
              {
                "affiliations": [
                  [
                    2,
                    "Lopez Ltd"
                  ]
                ],
                "identifiers": {
                  "orcid": "0000-0002-1825-0097"
                },
                "type": "personal",
                "name": "Robert Brown"
              },
```

**Comment by Glignos** [comment]
Screenshot 2020-11-10 at 11 36 59

**Pull request merged**